src/Uno.UI/UI/Xaml/UIElement.Pointers.Skia.cs

```diff
@@ -60,186 +107,402 @@ private void CoreWindow_PointerExited(CoreWindow sender, PointerEventArgs args)
+							// we continue to enumerate sibling children to detect a potential stale root.
+
+							while (child.MoveNext())
+							{
+								if (isChildStale(child.Current))
+								{
+									stale = SearchDownForStaleBranch(child.Current, isChildStale);
+									break;
+								}
+							}
 						}
 
-						raiseEvent(element);
-						return true;
+						TRACE($"> found child: {childResult.element.GetDebugName()} | stale branch: {stale?.ToString() ?? "-- none --"}");
+						return (childResult.element, stale);
 					}
 				}
+
+				// We didn't find any child at the given position, validate that element can be touched (i.e. not HitTestVisibility.Invisible),
+				// and the position is in actual bounds (which might be different than the clipping bounds)
+				if (elementHitTestVisibility == HitTestVisibility.Visible && renderingBounds.Contains(posRelToElement))
+				{
+					TRACE($"> LEAF! ({element.GetDebugName()} is the OriginalSource) | stale branch: {stale?.ToString() ?? "-- none --"}");
+					return (element, stale);
+				}
 				else
 				{
-					bool RecursePointerExited(UIElement e)
+					// If no stale element found yet, validate if the current is stale.
+					// Note: no needs to search down for stale child, we already did it!
+					if (isStale?.Invoke(element) ?? false)
 					{
-						if (e.IsOver(pointer))
-						{
-							foreach(var child in e.GetChildren().Reverse().ToArray())
-							{
-								if (RecursePointerExited(child))
-								{
-									return true;
-								}
-							}
+						stale = new Branch(element, stale?.Leaf ?? element);
+					}
 
-							e.OnNativePointerExited(pointerArgs);
-							return true;
-						}
+					TRACE($"> NOT FOUND (HitTestVisibility.Invisible or out of the **render** bounds) | stale branch: {stale?.ToString() ?? "-- none --"}");
+					return (default, stale);
+				}
+			}
 
-						return false;
+			private static Branch SearchDownForStaleBranch(UIElement staleRoot, Predicate<UIElement> isStale)
+				=> new Branch(staleRoot, SearchDownForStaleLeaf(staleRoot, isStale));
+
+			private static UIElement SearchDownForStaleLeaf(UIElement staleRoot, Predicate<UIElement> isStale)
+			{
+				foreach (var child in staleRoot.GetChildren().Reverse())
+				{
+					if (isStale(child))
+					{
+						return SearchDownForStaleLeaf(child, isStale);
 					}
+				}
+
+				return staleRoot;
+			}
+
+			#region Helpers
+			private static Func<IEnumerable<UIElement>, IEnumerable<UIElement>> Except(UIElement element)
+				=> children => children.Except(element);
+
+			private static Func<IEnumerable<UIElement>, IEnumerable<UIElement>> SkipUntil(UIElement element)
+				=> children => SkipUntilCore(element, children);
+
+			private static IEnumerable<UIElement> SkipUntilCore(UIElement element, IEnumerable<UIElement> children)
+			{
+				using var enumerator = children.GetEnumerator();
+				while (enumerator.MoveNext() && enumerator.Current != element)
+				{
```

Comment:
Codacy found an issue: [Either remove or fill this block of code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6213625)